### ISSUE TITLE
Manager says server accepted 0 of 20 items every second

### ISSUE THREAD

**Issue by binarypearl** [armadillica/flamenco#202]
In the output of the manager, I'm seeing this message every second:
`TaskUpdatePusher: server accepted 0 of 20 items.`

In the jobs view of the manager on blender cloud, there is one job that is in status 'CANCEL-REQUESTED' status, but that's not going away on it's own.

How did I get in this situation?  I had the first worker working.  I then tried to add a 2nd worker.  While probably not the best idea, I put the blender binary, flamenco worker, etc all on the same nfs share.  My goal was that they all just run in memory and then I only have to only have to maintain one copy.  After I kicked off the render with 2 workers, I noticed the job would go through but none of the bubbles were going blue/green for being rendered.  They were just staying grey.  I tried several other times with just one worker, but eventually got into this situation with the job pending cancellation and the manager with the server accepted 0 of 20 messages.

Do you know how I can recover from this situation?

--Shaun

**Comment by binarypearl** [comment]
I broke it further.  I tried to troubleshoot myself since I haven't use mangodb before and I figure why not?

I ended up trying to drop the flamanger database in mongo.  My hope was that I could bring everything to a clean slate and start over. 

So the 'server accepted 0 of 20 messages' is gone.  Now when I try to start a worker:
`2017-06-22 23:06:35,269  WARNING flamenco_worker.worker.FlamencoWorker Unable to POST to manager /sign-on, retrying in 30 seconds: 401 Client Error: Unauthorized for url: http://192.168.1.202:8083/sign-on`

And on the Manager:
`WARN[2017-06-22T23:06:05-04:00] Error fetching hashed password: not found`

I tried uninstalling mongo, removing the flamenco-worker.db database, wiping out and re-extracting the manager.

Basically at this point want to try to get to a clean slate and try again.

**Comment by sybrenstuvel** [comment]
> In the output of the manager, I'm seeing this message every second:
TaskUpdatePusher: server accepted 0 of 20 items.

In our server logs we can see that your Manager is trying to update tasks that don't exist. You probably archived the job while the Manager still had task updates queued for it.

> In the jobs view of the manager on blender cloud, there is one job that is in status 'CANCEL-REQUESTED' status, but that's not going away on it's own.

That means that the Manager had claimed tasks on that job for execution. Run the Manager to actually cancel those tasks at the Workers, and let it update the task status to "Canceled".

> While probably not the best idea, I put the blender binary, flamenco worker, etc all on the same nfs share.

That's fine, we do that too here in the studio. Just make sure the workers have a different working directory; otherwise they'll try and access the same SQLite database twice.

>  After I kicked off the render with 2 workers, I noticed the job would go through but none of the bubbles were going blue/green for being rendered. They were just staying grey. I tried several other times with just one worker, but eventually got into this situation with the job pending cancellation and the manager with the server accepted 0 of 20 messages.

Currently your project does show completed jobs, though.

> Do you know how I can recover from this situation?

Use `flamenco-manager -cleanslate` to erase all locally cached tasks. This won't help you with the 'server accepted 0 of 20 messages' message, though. For that I just added the `-purgequeue` CLI option, which I will soon release release as Flamenco Manager 2.0.12.

> I ended up dropping the flamanger database in mongo. My hope was that I could bring everything to a clean slate and start over.

This is a bad idea; it also erases the worker information from the database, causing the authentication errors you've seen.

> I tried uninstalling mongo, removing the flamenco-worker.db database, wiping out and re-extracting the manager.

This all won't help. The Workers have negotiated a username & password with the Manager, but the Manager has forgotten about this. The workers have to re-register, which you can accomplish by erasing the `worker_id` and `worker_secret` keys from their `~/.flamenco-worker.cfg` files.

The upcoming Worker version 2.0.6 now has a `--reregister` CLI option to make this easier.

**Comment by sybrenstuvel** [comment]
You can download the new versions from https://www.flamenco.io/download/, I just published them.

**Comment by sybrenstuvel** [comment]
>> In the output of the manager, I'm seeing this message every second:
>> TaskUpdatePusher: server accepted 0 of 20 items.
>
> In our server logs we can see that your Manager is trying to update tasks that don't exist. You probably archived the job while the Manager still had task updates queued for it.

I've updated Flamenco Server to make sure this doesn't cause task updates to hang any more (see [d8b96af3a](https://developer.blender.org/rFd8b96af3ae4b4a0b02ab518203266fc8f71762df)).

**Comment by binarypearl** [comment]
I'm back to one node working!

I wiped out the worker_id and worker_secret in the ~/.flamenco-worker.cfg file (I wondered where you were hiding that info :) )  Explains why when I tried removing and re-installing the worker that the issue was still happening.

I'm now running manager 2.0.12.  After doing the flamenco-manager -cleanslate, then everything started working again.

Question on the work directory comment above:
> That's fine, we do that too here in the studio. Just make sure the workers have a different working directory; otherwise they'll try and access the same SQLite database twice.

Not sure which setting you are referring to here:
There are the path_replacement variables in flamenco-manager.yaml

and in the Blender add on:
Job File Path:
Job Output Path:

of those, which needs to be unique for the worker?  Thanks!

--Shaun

**Comment by sybrenstuvel** [comment]
> of those, which needs to be unique for the worker? Thanks!

The [current working directory](https://en.wikipedia.org/wiki/Working_directory) is none of those; it's the "current directory" of every process on a computer, and typically it's the directory you start the process from.

**Comment by binarypearl** [comment]
Got it, thanks for all of your help!  Hopefully now I'll be able to get 2 nodes going successfully.  I have family coming into town so I'll have limited time to work on this coming week.  But I'm happy with the progress that we made.

fyi:  My 2nd node is another Linux system.  Once I get comfortable that it's working, I'm planning on getting a new video card/power supply for my Windows computer so I'll have 3 nodes.  And then possibly a work laptop that has an nvidia quadro m1000m.  With one quick render I'm guessing it would be able to render about 40-45 frames in what my 780 could do in 100 frames.

--Shaun

**Comment by sybrenstuvel** [comment]
Have a nice weekend, and keep us updated!
